yt_dlp/extractor/safari.py

```diff
@@ -150,6 +151,10 @@ def _real_extract(self, url):
             'flashvars[referenceId]': reference_id,
         }
 
+        if not self.LOGGED_IN:
+            if self.is_logged():
+                self.LOGGED_IN = True
+
         if self.LOGGED_IN:
```

Comment:
Needless. It's enough to set `self.LOGGED_IN` it in `_perform_login`.

Reply:
Not true. That is the hole point of this change and the #1027
If you only set in perform_login cookies are not used if you dont use username + password.

This PR is to check, using cookies, if the user is logged in and to use cookies properly. That is the only option for SSO users that cannot use username + password login.

Reply:
Understood, and I completely agree: df6e7e70f2469409f877943ff03cfabf03667a11.
It gets set in `_initialize_pre_login`.

Reply:
Is `_initialize_pre_login` called if user only uses cookies?

Try download full videos ( not only first minute ) using your code and using cookies. If it works than this code is already merged in your PR. This PR and the #1027 will be closed as well.

Reply:
I believe that's correct. Could you check?
It downloads a ~6:00 long video.
[create-cookie-output.txt](https://github.com/yt-dlp/yt-dlp/files/8603303/create-cookie-output.txt)
[output.txt](https://github.com/yt-dlp/yt-dlp/files/8603304/output.txt)

Reply:
Yea, it is working. GJ

```
C:\yt-dlp-3dcac0e131bf3d58fae9c2362c1c526594ff2dcf>python -m yt_dlp --cookies-from-browser chrome "https://learning.oreilly.com/videos/mastering-jmeter-5-0/9781838640880/"
[Cookies] Extracting cookies from chrome
[Cookies] Extracted 1143 cookies from chrome
[safari:course] Checking if logged in
[safari:course] 9781838640880: Downloading course JSON
[download] Downloading playlist: Mastering JMeter 5.0
[safari:course] playlist Mastering JMeter 5.0: Collected 52 videos; downloading 52 of them
[download] Downloading video 1 of 52
[safari:api] Checking if logged in
[safari:api] 9781838640880/video1_1: Downloading part JSON
[safari] Checking if logged in
[safari] 9781838640880-video1_1: Downloading kaltura session JSON
[Kaltura] 9781838640880-video1_1: Downloading webpage
[Kaltura] 0_j24uni29: Downloading video info JSON
[Kaltura] 0_j24uni29: Checking mp4-521 URL
[Kaltura] 0_j24uni29: Downloading m3u8 information
[info] 0_j24uni29: Downloading 1 format(s): mp4-1116
[download] Destination: The Course Overview [0_j24uni29].mp4
[download] 100% of 36.29MiB in 00:04
[download] Downloading video 2 of 52
[safari:api] 9781838640880/video1_2: Downloading part JSON
[safari] 9781838640880-video1_2: Downloading kaltura session JSON
[Kaltura] 9781838640880-video1_2: Downloading webpage
[Kaltura] 0_bam6t353: Downloading video info JSON
[Kaltura] 0_bam6t353: Checking mp4-532 URL
[Kaltura] 0_bam6t353: Downloading m3u8 information
[info] 0_bam6t353: Downloading 1 format(s): mp4-532
[download] Destination: Duration and Size Assertions [0_bam6t353].mp4
[download] 100% of 50.28MiB in 00:03
[download] Downloading video 3 of 52
[safari:api] 9781838640880/video1_3: Downloading part JSON
[safari] 9781838640880-video1_3: Downloading kaltura session JSON
[Kaltura] 9781838640880-video1_3: Downloading webpage
[Kaltura] 0_ryp0vswg: Downloading video info JSON
[Kaltura] 0_ryp0vswg: Checking mp4-524 URL
[Kaltura] 0_ryp0vswg: Downloading m3u8 information
[info] 0_ryp0vswg: Downloading 1 format(s): mp4-537
[download] Destination: HTML Assertions [0_ryp0vswg].mp4
[download] 100% of 15.50MiB in 00:02
[download] Downloading video 4 of 52
[safari:api] 9781838640880/video1_4: Downloading part JSON
[safari] 9781838640880-video1_4: Downloading kaltura session JSON
[Kaltura] 9781838640880-video1_4: Downloading webpage
[Kaltura] 0_a5wqpxif: Downloading video info JSON
[Kaltura] 0_a5wqpxif: Checking mp4-533 URL
[Kaltura] 0_a5wqpxif: Downloading m3u8 information
[info] 0_a5wqpxif: Downloading 1 format(s): mp4-566
[download] Destination: XML and XML Schema Assertions [0_a5wqpxif].mp4
[download] 100% of 33.48MiB in 00:03
[download] Downloading video 5 of 52
[safari:api] 9781838640880/video1_5: Downloading part JSON
[safari] 9781838640880-video1_5: Downloading kaltura session JSON
[Kaltura] 9781838640880-video1_5: Downloading webpage
[Kaltura] 0_2fuux6mp: Downloading video info JSON
[Kaltura] 0_2fuux6mp: Checking mp4-532 URL
[Kaltura] 0_2fuux6mp: Downloading m3u8 information
[info] 0_2fuux6mp: Downloading 1 format(s): mp4-594
[download] Destination: Response Assertion [0_2fuux6mp].mp4
[download] 100% of 25.71MiB in 00:03
[download] Downloading video 6 of 52
[safari:api] 9781838640880/video1_6: Downloading part JSON
[safari] 9781838640880-video1_6: Downloading kaltura session JSON
[Kaltura] 9781838640880-video1_6: Downloading webpage
[Kaltura] 0_uppk90n6: Downloading video info JSON
[Kaltura] 0_uppk90n6: Checking mp4-533 URL
[Kaltura] 0_uppk90n6: Downloading m3u8 information
[info] 0_uppk90n6: Downloading 1 format(s): mp4-657
[download] Destination: XPath and JSON Assertions [0_uppk90n6].mp4
[download] 100% of 87.99MiB in 00:07
[download] Downloading video 7 of 52
[safari:api] 9781838640880/video1_7: Downloading part JSON
[safari] 9781838640880-video1_7: Downloading kaltura session JSON
[Kaltura] 9781838640880-video1_7: Downloading webpage
[Kaltura] 0_ub2qwmkb: Downloading video info JSON
[Kaltura] 0_ub2qwmkb: Checking mp4-521 URL

ERROR: Interrupted by user
```